### ISSUE TITLE
strict access control and mutability annotations

### ISSUE THREAD

**Issue by bryanmontz** [planetary-social/nos#1677]
We should make our access control annotations (private, private(set), or not) and our mutability annotations (let vs var) as strict as possible.

Why?
1. Cognitive load: These annotations are not only for the compiler; they make it easier to reason about what ought or ought not to happen. They help coders accurately describe data, data flows, and user flows according to Product requirements, and they help current coders prevent future bugs.
2. Compiler optimization: If the compiler knows something is immutable, it can optimize for faster execution.

For access controls, coders should ask themselves with each API (property, function, class, etc): Do outsiders need to have access to this information or this function ("need to know" principle)? If not, it should be marked `private` (or sometimes `fileprivate` as the case may be). Or maybe outsiders need only read access and not write access (`private(set)`).

In short:
1. If it could be `private`, it should be `private`.
2. If it could be immutable, it should be immutable.
